### PR TITLE
fix bfill handler for supernova

### PR DESCRIPTION
**Pull request by catfact**
(new version of PR #4137, based on 3.10 branch)

Purpose and Motivation
----------------------

noticed that `/b_fill` commands didn't seem to be doing anything in supernova. 

checking the source revealed that the sample index into the buffer data wasn't being incremented, so i added this.

i additionally added a bounds check for the index. surprisingly this doesn't happen in other command handlers like `/b_set`. it is possible to smash memory and crash supernova using these OSC commands.

Types of changes
----------------

- Bug fix (non-breaking change which fixes an issue)

Checklist
---------

- [x] All previous tests are passing
- [x] This PR is ready for review

i haven't added any unit tests, but of course would be happy to do so with a little more guidance.

### CODE REVIEW

**Comment by catfact** [comment]
nevermind, didn't realize i could change base branch for existing PR.

**Comment by catfact** [comment]
... aaand, nevermind, GH is not happy with that either, it appears to use some cached version of the branch and makes the rebase pointless.

File: server/supernova/sc/sc_osc_handler.cpp [hunk 1]
```diff
@@ -2740,20 +2740,27 @@ void handle_b_fill(ReceivedMessage const & msg)
...
         osc::int32 index = it->AsInt32(); ++it;
         verify_argument(it, end);
         osc::int32 samples = it->AsInt32(); ++it;
         verify_argument(it, end);
         float value = extract_float_argument(it++);
 
-        for (int i = 0; i != samples; ++i)
+        for (int i = 0; i != samples; ++i) {
             data[index] = value;
+	    ++index;
```
Comment (patrickdupuis) [inline server/supernova/sc/sc_osc_handler.cpp]: Please replace the tab with spaces here.

File: server/supernova/sc/sc_osc_handler.cpp [hunk 2]
```diff
@@ -2740,20 +2740,27 @@ void handle_b_fill(ReceivedMessage const & msg)
...
+            if (index >= bufSamples) {
+	        break;
```
Comment (patrickdupuis) [inline server/supernova/sc/sc_osc_handler.cpp]: Same thing here.

**Comment by brianlheim** [comment]
@catfact thanks, please address indentation issues as req'd by @patrickdupuis 

unit tests for this would be nice but we don't currently have a good system in place for that, so they aren't required for this PR

**Comment by catfact** [comment]
pardons folks, i am too used to auto-formatting, have untabifyed. 
truly all i wished to do was point out a silly bug and a simple fixed.

**Comment by brianlheim** [comment]
@catfact Thanks! Sorry it took so long to get this through. Definitely consider setting your editor to show spaces/tabs, it helps a lot. And if you need help with rebasing just ask, it can be a struggle sometimes with GH's UI but there is usually a clean way to resolve things if you get stuck.

**Comment by catfact** [comment]
@brianlheim, this is basically OT but for the record
> consider setting your editor to show spaces/tabs

writing c/c++ is my day job and different repos have different requirements. my company uses tabs so that is my basic setting for emacs c++mode.

for supercollider fork i've set a git pre-commit hook to untabify and apply a specific clang-format. so hopefully you will have no more format issues from me.

GH's PR behavior is another kettle of fish. i swear it became quite confused with rebasing an open PR as suggested.

**Comment by brianlheim** [comment]
> for supercollider fork i've set a git pre-commit hook to untabify and apply a specific clang-format. so hopefully you will have no more format issues from me, if and when i have future contributions

@catfact You'll hopefully be happy to learn that at this very moment we are adding a clang-format file to this repo, applying it across the entire codebase, and adding it as a CI linting step.

**Pull request merged**